velox/functions/sparksql/SplitFunctions.cpp

```diff
@@ -134,10 +135,128 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
               .build()};
 }
 
+class StringToMap final : public exec::VectorFunction {
+ public:
+  StringToMap(char entryDelim, char keyValueDelim)
+      : entryDelim_(entryDelim), keyValueDelim_(keyValueDelim) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    DecodedVector* inputString = decodedArgs.at(0);
+
+    BaseVector::ensureWritable(
+        rows, MAP(VARCHAR(), VARCHAR()), context.pool(), result);
+    exec::VectorWriter<Map<Varchar, Varchar>> resultWriter;
+    resultWriter.init(*result->as<MapVector>());
+
+    folly::F14FastMap<StringView, vector_size_t> keyToIdx;
+    rows.applyToSelected([&](vector_size_t row) {
+      resultWriter.setOffset(row);
+      auto& mapWriter = resultWriter.current();
+
+      const StringView& current = inputString->valueAt<StringView>(row);
+      const char* pos = current.begin();
+      const char* end = pos + current.size();
+      const char* entryEnd;
+      const char* keyEnd;
+      do {
+        entryEnd = std::find(pos, end, entryDelim_);
+        keyEnd = std::find(pos, entryEnd, keyValueDelim_);
+        auto key = StringView(pos, keyEnd - pos);
+        auto iter = keyToIdx.find(key);
+        VELOX_USER_CHECK(
```

Comment:
VELOX_USER_CHECK_EQ

Reply:
Compiler complains type mismatched if we use it. So keep it unchanged.